hera_cal/smooth_cal.py

```diff
@@ -105,9 +106,109 @@ def time_filter(gains, wgts, times, filter_scale=1800.0, nMirrors=0):
     return conv_gains[nBefore: nBefore + len(times), :]
 
 
+def time_freq_2D_filter(gains, wgts, freqs, times, freq_scale=10.0, time_scale=1800.0, 
+                        tol=1e-09, filter_mode='rect', maxiter=100, window='tukey', **win_kwargs):
+    '''Filter calibration solutions in both time and frequency simultaneously. First rephases to remove
+    a time-smoothed delay from the gains, then performs the low-pass 2D filter in time and frequency,
+    then puts back in the delay rephasor. Uses aipy.deconv.clean to account for weights/flags.
+
+    Arguments:
+        gains: ndarray of shape=(Ntimes,Nfreqs) of complex calibration solutions to filter
+        wgts: ndarray of shape=(Ntimes,Nfreqs) of real linear multiplicative weights
+        freqs: ndarray of frequency channels in Hz
+        times: ndarray of shape=(Ntimes) of Julian dates as floats in units of days
+        freq_scale: frequency scale in MHz to use for the low-pass filter. freq_scale^-1 corresponds
+            to the half-width (i.e. the width of the positive part) of the region in fourier
+            space, symmetric about 0, that is filtered out.
+        time_scale: time scale in seconds. Defined analogously to freq_scale.
```

Comment:
Confusingly, times is in days (because that's the units in a UVData object), but time_scale is in seconds. I've added a couple of notes to the documentation to highlight this fact.